src/python/torchdistx/deferred_init.py

```diff
@@ -35,11 +35,11 @@ def deferred_init(module_fn: Callable[..., Module], *args, **kwargs) -> Module:
         to a module after its returned from ``deferred_init()``; otherwise it
         cannot be correctly materialized.
     """
-    _C.enable_deferred_init(True)
+    _C.enter_deferred_init()
```

Comment:
shall we put `enter_deferred_init`, `leave_deferred_init` into context manager?

Reply:
Well that has been debated quite a bit back when I was designing the API. We are right now artificially limiting this feature to an explicit `deferred_init()` function in order to reduce the test surface, but technically you are right. The natural fit for this function pair would be a context manager. Maybe we can consider expanding our scope in the future.